src/main/java/org/opensearch/flowframework/FlowFrameworkPlugin.java

```diff
@@ -185,9 +187,23 @@ public List<ExecutorBuilder<?>> getExecutorBuilders(Settings settings) {
             new ScalingExecutorBuilder(
                 WORKFLOW_THREAD_POOL,
                 1,
-                OpenSearchExecutors.allocatedProcessors(settings),
-                TimeValue.timeValueMinutes(5),
+                Math.max(1, OpenSearchExecutors.allocatedProcessors(settings) - 1),
+                TimeValue.timeValueMinutes(1),
                 FLOW_FRAMEWORK_THREAD_POOL_PREFIX + WORKFLOW_THREAD_POOL
+            ),
+            new ScalingExecutorBuilder(
+                PROVISION_WORKFLOW_THREAD_POOL,
+                1,
+                Math.max(1, OpenSearchExecutors.allocatedProcessors(settings) - 1),
+                TimeValue.timeValueMinutes(5),
+                FLOW_FRAMEWORK_THREAD_POOL_PREFIX + PROVISION_WORKFLOW_THREAD_POOL
+            ),
+            new ScalingExecutorBuilder(
```

Comment:
I feel like we are having couple of `ScalingExecutorBuilder` for our plugin. 
Even [ml-commons](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/plugin/MachineLearningPlugin.java#L772) has FixedExecutorBuilders because of which we have to handle the flaky tests. 
IMO we should make `PROVISION_WORKFLOW_THREAD_POOL` and `DEPROVISION_WORKFLOW_THREAD_POOL` as `FixedExecutorBuilders` to avoid spawning of threads.

Reply:
Fixed executors will still spawn the threads in exactly the same way Scaling will, the difference is that all the spawned threads will stay alive and idle at the max.    

WIth scaling, after threads are idle for the time value, they will shut down.    

Pictorially, assuming 4 processors the max size is 3.

Fixed:   1 -> 2 -> 3 -> stays forever with 3 idle threads
Scaling: 1 -> 2 -> 3 -> stays for timeout of oldest thread -> 2 -> stays for timeout of oldest thread -> 1 idle thread

Reply:
WIth fixed we have to set a max queue size and it will fail there.  With scaling the queue can grow indefinitely.

Reply:
AD uses ScalingExecutorBuilders. They are recommended for non-steady workloads, which Flow Framework is.  It will be idle most of the time and thread pools should release unused idle threads.

Fixed Executor Builders are used for predictable steady-state workloads where threads need to always be ready to execute.